modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryIndexing.java

```diff
@@ -407,4 +407,27 @@ public void remove(GridCacheContext cctx, GridQueryTypeDescriptor type, CacheDat
      * @throws IgniteCheckedException On bean registration error.
      */
     void registerMxBeans(IgniteMBeansManager mbMgr) throws IgniteCheckedException;
+
+    /**
+     * Return table information filtered by given patterns.
+     *
+     * @param schemaNamePtrn Filter by schema name. Can be {@code null} to don't use the filter.
+     * @param tblNamePtrn Filter by table name. Can be {@code null} to don't use the filter.
+     * @param tblTypes Filter by table type. As Of now supported only 'TABLES' and 'VIEWS'.
+     * Can be {@code null} or empty to don't use the filter.
+     *
+     * @return Column information filtered by given patterns.
+     */
+    Collection<TableInformation> tablesInformation(String schemaNamePtrn, String tblNamePtrn, String... tblTypes);
```

Comment:
Who do we place `tablesInformation` and `columnsInformation` into `GridQueryIndexing`? The interface (and the implementation) is dense enough. Could we separate the concerns and those methods outside? `SchemaManager` looks like a good candidate.

Reply:
As of now SchemaManager it's internal part of indexing. We need to use information provided by new methods in core module. Earlier I talked with VOzerov about the same and we decided expose the new methods in GridQueryIndexing.

Reply:
Yes, perhaps we need to place methods to `GridQueryIndexing`. But should we place an implementation into `IgniteH2Indexing`? Would not it be better (and why) to place implementation into `SchemaManager` and just delegate method call in `IgniteH2Indexing`?

Reply:
I'm not sure that's right way. SchemaManager provide simple information about schema. IgniteH2Indexing can enrich all other required information.